### PR TITLE
Add timeout by dataset size

### PR DESCRIPTION
**Pull request by XuanYang-cn**

### CODE REVIEW

**Comment by XuanYang-cn** [comment]
image

File: vectordb_bench/backend/cases.py [hunk 1]
```diff
@@ -88,6 +88,20 @@ def filters(self) -> dict | None:
...
+        else:
+            return 60 * 60 * 24 # 24h
+
+    @property
+    def optimize_timeout(self) -> int:
+        if self.dataset.data.size <= 1_000_000:
+            return 60 * 60 * 1 # 1h
+        else:
+            return 60 * 60 * 8 # 8h
+
```
Comment (liliu-z) [inline vectordb_bench/backend/cases.py:104]: Can we binding timeout with specific case? Since different dataset, dim can also affect the performance

**Review by liliu-z — CHANGES_REQUESTED** [review]

**Review by liliu-z — APPROVED** [review]

File: vectordb_bench/models.py [hunk 2]
```diff
@@ -167,20 +167,21 @@ def append_return(x, y):
         max_qps = max(map(len, [str(f.metrics.qps) for f in filtered_results])) + 3
         max_recall = max(map(len, [str(f.metrics.recall) for f in filtered_results])) + 3
 
-        max_db_labels = 8 if max_db_labels == 0 else max_db_labels
-        max_load_dur = 11 if max_load_dur == 0 else max_load_dur + 3
-        max_qps = 10 if max_qps == 0 else max_load_dur + 3
-        max_recall = 13 if max_recall == 0 else max_recall + 3
+        max_db_labels = 8 if max_db_labels < 8 else max_db_labels
+        max_load_dur = 11 if max_load_dur < 11 else max_load_dur
+        max_qps = 10 if max_qps < 10 else max_load_dur
```
Comment (alwayslove2013) [inline vectordb_bench/models.py:172]: if qps < 10 then qps = load_dur ...
spelling error?

**Review by alwayslove2013 — APPROVED** [review]

**Pull request merged**